workflow-manager/bucket/bucket.go

```diff
@@ -132,17 +181,26 @@ func (b *Bucket) listFilesGS(ctx context.Context) ([]string, error) {
 		return nil, fmt.Errorf("storage.newClient: %w", err)
 	}
 
+	return client, nil
+}
+
+func (b *Bucket) listFilesGS(ctx context.Context) ([]string, error) {
+	client, err := b.gcsClient(ctx)
```

Comment:
I recommend creating a GCSService calls that lives across the lifetime of the process and holds the GCS client. This way you don't need to create a new client for every GCS call (which is inefficient). The client can be long lived.